src/libraries/Microsoft.PowerFx.Core/Public/Engine.cs

```diff
@@ -173,92 +201,115 @@ public CheckResult Check(string expressionText, RecordType parameterType, Parser
         /// <param name="parameterType">types of additional args to pass.</param>
         /// <param name="options">parser options to use.</param>
         /// <returns></returns>
-        public CheckResult Check(ParseResult parse, RecordType parameterType, ParserOptions options = null)
+        [Obsolete("Use other check overload. Shouldn't need both ParserOptions and ParseResult.")]
+        public CheckResult Check(ParseResult parse, RecordType parameterType, ParserOptions options)
+        {
+            return Check(parse, parameterType);
+        }
+
+        [Obsolete("Use other check overload. Shouldn't need both ParserOptions and ParseResult.")]
+        internal CheckResult Check(ParseResult parse, ParserOptions options)
+        {
+            parse.Options = options;
+            return Check(parse, null, options);
+        }
+
```

Comment:
We should remove these APIs now

Reply:
Eventually - but keeping them since this is "just" a refactor and I want to avoid PAClient breaking changes.